src/QirTools/pyqir/qirlib/src/interop.rs

```diff
@@ -0,0 +1,121 @@
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct QuantumRegister {
+    pub name: String,
+    pub index: u64,
+}
+
+impl QuantumRegister {
+    pub fn new(name: String, index: u64) -> Self {
+        QuantumRegister { name, index }
+    }
+
+    pub fn as_register(&self) -> Register {
+        Register::Quantum(self.clone())
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ClassicalRegister {
+    pub name: String,
+    pub size: u64,
+}
+
+impl ClassicalRegister {
+    pub fn new(name: String, size: u64) -> Self {
+        ClassicalRegister { name, size }
+    }
+
+    pub fn as_register(&self) -> Register {
+        Register::Classical(self.clone())
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Register {
+    Quantum(QuantumRegister),
+    Classical(ClassicalRegister),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Controlled {
```

Comment:
Can this be used to control arbitrary instructions (similar to the `Controlled` keyword in Q#)? If not, what instruction does this represent?